File: mobile/src/main/java/com/alexstyl/specialdates/upcoming/YearHeaderViewModel.java
package com.alexstyl.specialdates.upcoming;

public final class YearHeaderViewModel implements UpcomingRowViewModel {
    private final String year;

    YearHeaderViewModel(String year) {
        this.year = year;
    }

    @Override
    public int getViewType() {
        return UpcomingRowViewType.YEAR;
    }

    public String getYear() {
        return year;
    }
}

File: android_mobile/src/main/java/com/alexstyl/specialdates/upcoming/YearHeaderViewModel.kt
package com.alexstyl.specialdates.upcoming

data class YearHeaderViewModel internal constructor(val year: String) : UpcomingRowViewModel {

    override val viewType: Int
        get() = UpcomingRowViewType.YEAR

    override val id: Long
        get() = hashCode().toLong()
}
